### ISSUE TITLE
[EKS Windows] [EBS CSI Driver]: Managed EBS CSI Driver Add-On for windows 2022

### ISSUE THREAD

**Issue by IftequarWin** [aws/containers-roadmap#1886]
**Tell us about your request**
What do you want us to build?

- Please provide the AMazon managed Add-Ons for the EBS CSI driver for windows 2022 as well

**Which service(s) is this request for?**

- EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

- We want to use EBS CSI Driver for windows but have to install it using the helm but we want the solution to add it as a Amazon managed Add-On as in Linux. Reference : https://docs.aws.amazon.com/eks/latest/userguide/managing-ebs-csi.html

**Are you currently working around this issue?**
How are you currently solving this problem?

- We are solving it by installing it using the helm. Reference : https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/examples/kubernetes/windows

**Comment by mikestef9** [comment]
This will be available add-on version 1.12.1-eksbuild.1 which is working its way through deployment process and released soon.

**Comment by youwalther65** [comment]
We have rolled out add-on versions EBS CSI v1.13.0-eksbuild.2 (default version now for EKS v1.22, v1.23 and v1.24) and v1.12.1-eksbuild.2  in all regions. See [here ](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/Dockerfile#L45-47). Please check!

**Comment by jans-jeroen** [comment]
I've tried upgrading our testing cluster to the addon version V1.13.0-eksbuild.2.
The EBS CSI driver image (`602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/aws-ebs-csi-driver:v1.13.0`) does have an image for Windows server 2022 present.

The other images however no longer have a Windows Server 2022 image:
- `602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6`
- `602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6`

This is on an EKS 1.24 cluster with Windows Server 2022 nodes (AMI: `ami-0d4fc8ef7717c70d6`, kernel-version: `10.0.20348.1249`) in the region `eu-west-1`

The events from the related pod (`kubectl describe pod -n kube-system ebs-csi-node-windows-mg6zl`):

```
  Normal   Pulled             16s   kubelet                  Container image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/aws-ebs-csi-driver:v1.13.0" already present on machine
  Normal   Created            16s   kubelet                  Created container ebs-plugin
  Warning  Failed             4s    kubelet                  Failed to pull image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6": rpc error: code = NotFound desc = failed to pull and unpack image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6": no match for platform in manifest: not found
  Normal   Pulling            4s    kubelet                  Pulling image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6"
  Normal   Started            4s    kubelet                  Started container ebs-plugin
  Warning  Failed             4s    kubelet                  Error: ErrImagePull
  Normal   Pulling            4s    kubelet                  Pulling image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6"
  Warning  Failed             3s    kubelet                  Failed to pull image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6": rpc error: code = NotFound desc = failed to pull and unpack image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6": no match for platform in manifest: not found
  Normal   BackOff            3s    kubelet                  Back-off pulling image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6"
  Normal   BackOff            3s    kubelet                  Back-off pulling image "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6"
```

The old liveness probe image (`602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.6.0-eksbuild.0`) does have an image for Windows Server 2022 as shown by `docker manifest inspect 602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.6.0-eksbuild.0`:

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:b8239acbd0eea0b11373547052882543ae3c5ec907213bc9815a862d49cd3df7",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:9d04324a06128c639100911faad7b126838df3957455c81aa13d662dafb9a119",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.3406"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:f9df3af9ece6253a4b9765bedf206544357c399216510c6c87f6376ac2f2b593",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18363.1556"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:fa0d5478229f67d6dc09cc8278e48a1d68aabb901680c5b21b811d22dca9159f",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19041.1415"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:ff9742d4bfd4e6f5b407632a515ee51fc763f057796083e6e629bfadc4bc4384",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19042.1889"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:5d8e61f0d43d975f7a70a9535a1f8b6a15bb32b894caf38440203162eb7f66c4",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.20348.1006"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:72f1c4b03d62201d313beb05fa62e5875255124b0c2836c126cd5df4ec492977",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:576506a92bcbf0c2711f227ebfab4eb6c84bd64ec3861a9894f0edf371e5ee0f",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:a40537a4d76c83fd2481de6f63828ef92e0655eab92071d60552992a14c62822",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:8224e04aeb19d8af394f94f978215a2ca82ef9082fdeffb17f13411890d2b84f",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:99d29cf859a6df60216ef0c640860768f21571ea2703a493f6bf01252f6ec597",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

The same goes for the CSI Node driver registrar image: `docker manifest inspect 602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eksbuild.0`

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:0d2ce74f7ccfc106f1cbfdadaf85ab8efc2007632cc4fb45e245e7c77267d8fb",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:79e01b38340862a35a125d9196650734c4761090ce2a7e2057259299f903f023",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.3406"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:c15551048e4cdd7280fa9a78ffb208f6284f40b9cc62c1cb7bff7c3361c81200",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18363.1556"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:bf5ac82ea1ff0862f92979193728005496a64061d49521479e1d871b28742d5e",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19041.1415"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:c091366d17522a1f0b818096622a7ebbbaccf3b013bc4eae51e1211ce04e1d33",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19042.1889"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 950,
         "digest": "sha256:190ee26cbd6f1a7d57258801769a86e7b7d9ba9aeb6d198d5d7b1157196a19b1",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.20348.1006"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:bdec4c2c9d0c9232312777924b7885db92b933257fbbb614bc80105714c0f28b",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:15257cba0b944365d986b8b2c6501b76db3c5e98266a6213aefa0f9bf19eb017",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:5f0cb84816b86355e16d289b759fef907bf82112870b3175d8275c56d9a9712c",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:a56420833a9cc311494febda84f0bc89bc80195b7a0e75c6da0c26739f370be6",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:36dc45ca001cbf394d44ab5934c53aa59b92a519bb8cf7a2497bfe71a30db6cb",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

The new images (Liveness and CSI node driver registrar) however only have Linux images available (checked using podman as docker on Windows was having issues to inspect the images):
- `podman manifest inspect 602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/livenessprobe:v2.7.0-eks-1-23-6`

```
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 1116,
            "digest": "sha256:cde75492ef6512d0442e8745dfa66f308214cf3d24610bd1372fcc247a21c76d",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 1116,
            "digest": "sha256:6c36030defb4e9148e4cb95c81d5826353a6d1ecf553586cf0012a9f2ee25288",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        }
    ]
}
```

- `podman manifest inspect 602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/csi-node-driver-registrar:v2.5.1-eks-1-23-6`

```
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 1116,
            "digest": "sha256:98b281d47fd6d91a5468109944531d805b1289d20089600c05447459b18979c3",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 1116,
            "digest": "sha256:06bddaa398e2332010d3d0ce63bd346b60066199cc418be1ddf3d3489d4009d3",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        }
    ]
}
```

**Comment by youwalther65** [comment]
@jans-jeroen  Thank you for the feedback and sorry to hear that! I will get in contact with our internal team immediately after the Thanksgiving weekend and follow-up here.

**Comment by ishaikh68** [comment]
@youwalther65 do you have any latest updates on this.

**Comment by youwalther65** [comment]
@ishaikh68 We haven't forget you but we were quite busy because of re:Invent. We are just in the rollout of the fix ( new add-on version) in all regions. I will post an update once this is successfully completed. Thank you for your patience

**Comment by torredil** [comment]
Hey all, the new release is now available. If you using the EKS add-on and deploying the driver on Windows nodes please upgrade to `v1.13.0-eksbuild.3`. The fix has also been back-ported to `v1.12.1-eksbuild.3` & `v1.11.5-eksbuild.2`.
